### ISSUE TITLE
Make a single window mode

### ISSUE THREAD

**Issue by codemac** [getting-things-gnome/gtg#275]
I use a tiling window manager, and I've been playing around with GTG. The popup windows that show task detail are very annoying for a tiling window manager user.

A dedicated separate detail window that I could resize and place somewhere, or having the app have a "single window mode" would be really welcome.

Seeing all the recent activity in the repo is so exciting! Thanks for all the hard work, and if there are any immediate work arounds please let me know :)

**Comment by diegogangl** [comment]
Yeah the popups can be annoying unless you want to keep a task open for reference or something.

My (vague) idea was to have a third pane in the task browser. So it would be `sidebar | tasks | task editor`. To do this we need to use a listbox in the browser first which is heavy refactoring and redesign work, so it could take a while. The task editor would also need some re-thinking.

The only workaround I can think of is using the quick entry and Ctrl+D as much as possible and only opening tasks when it's really necessary. 

Tentatively marking this for 0.5, depending on how long the other stuff takes

**Comment by nekohayo** [comment]
I've felt the annoyance with the "forced window management" of the current approach too, and I was _also_ thinking it could be nice to have an option for task editor instances to be "docked" and stacked inside a scrolling sidebar on the right (that's my "gut feeling" design hunch, but I haven't spent proper time thinking about it), which is exactly what Diego is suggesting there... but I'd say this would need to be an option, because it probably won't please (nor work for) everybody: it would "visually clutter up" the UI, and it might not work well with small screen resolutions where independent windows may sometimes work better.

In any case, it's the kind of thing I'd tend to say "someone needs to provide a design and patch for it" (aka "patch-or-wont-happen" label?)... I wouldn't necessarily assign it to a milestone unless Diego is super excited about working on that feature personally for 0.5

**Comment by diegogangl** [comment]
> and it might not work well with small screen resolutions where independent windows may sometimes work better.

Indeed, we should have a button to pop out the task into a separate window. As for the small screen resolutions, once we are using listboxes and meson we can look into libhandy (or whatever Gtk4 has).

> In any case, it's the kind of thing I'd tend to say "someone needs to provide a design and patch for it" (aka "patch-or-wont-happen" label?)... I wouldn't necessarily assign it to a milestone unless Diego is super excited about working on that feature personally for 0.5

I think we need to be more involved in this. It's going to need several changes that also work with the design changes we do to the task browser. We should iterate over some mockups and approve a design before any MRs happen

**Comment by nekohayo** [comment]
Here's a crazy idea for a hybrid approach: what if we could embed the task editor in a GTK [Popover](https://python-gtk-3-tutorial.readthedocs.io/en/latest/popover.html) widget (oriented sideways if there's empty space available on the right of GTG's main window, otherwise I guess oriented above or below) that shows up when selecting (single-clicking) a task?

Potential advantages:

* Probably quite a bit easier and less intrusive (in terms of code) to implement
* May be visually cleaner than "have a second sidebar shown at all times, filled with listboxes containing task editors"
* It eliminates window/boxes "management" because the popovers would close when you click on something else anyway

Disadvantages:

* Works best on screens where GTG is not maximized / where you have empty space on the side of the window.
* Focused on a single task at once (which may be considered a "feature" for some). For those who would want multiple things open at once, there would need to be an option to choose between the popover approach and the current "each task opens in a standalone window on double-click" behavior (or the potential "listboxes" replacement).

**Comment by nekohayo** [comment]
For what it's worth, if you scroll through https://wiki.gnome.org/Apps/GTG/Design you can also see someone once thought of the "show task contents in a sidebar" in a mockup: https://wiki.gnome.org/Apps/GTG/Design?action=AttachFile&do=get&target=gtg-mockup-workview.jpg (and https://wiki.gnome.org/Apps/GTG/Design/Old also had a bunch of funky things).

**Comment by diegogangl** [comment]
The dark sidebar on the last one looks really good man. I'm willing to fight the HIG for that one :)
